helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -333,6 +370,7 @@ public synchronized void refresh(HelixDataAccessor accessor) {
     _stateModelDefinitionCache.refresh(accessor);
     _clusterConstraintsCache.refresh(accessor);
     updateMaintenanceInfo(accessor);
+    timeoutNodesDuringMaintenance(accessor, _clusterConfig, _isMaintenanceModeEnabled);
```

Comment:
Any specific reason we do not put this into refreshLiveInstances()?

Reply:
Yes, because we want this step to be after updateMaintenanceInfo(), which makes the most logical sense.